consensus/istanbul/validator/weighted.go

```diff
@@ -96,6 +96,7 @@ func newWeightedValidator(addr common.Address, reward common.Address, votingpowe
 }
 
 type weightedCouncil struct {
+	chain             consensus.ChainReader
```

Comment:
When does this `chain` is used? I can't find any usage.

Reply:
It's in refreshProposer function.
but your ideal is better that you mentioned next comment.
so i will remove that, and fix liken `calcWeight()`.